tests/cnf/core/network/security/tests/nftables-custom-firewall-test.go

```diff
@@ -0,0 +1,505 @@
+	configMapData["daemons"] = daemonsConfig
+	configMapData["frr.conf"] = frrConfig
+	configMapData["vtysh.conf"] = vtyShConfig
+
+	return configMapData
+}
+
+func removePrefixFromIPList(ipAddressList []string) []string {
+	var ipAddressListWithoutPrefix []string
+	for _, ipaddress := range ipAddressList {
+		ipAddressListWithoutPrefix = append(ipAddressListWithoutPrefix, ipaddr.RemovePrefix(ipaddress))
+	}
+
+	return ipAddressListWithoutPrefix
+}
+
+func buildRoutesMapWithSpecificRoutes(podList []*pod.Builder, nextHopList []string) map[string]string {
+	Expect(len(podList)).ToNot(BeZero(), "Pod list is empty")
+	Expect(len(nextHopList)).ToNot(BeZero(), "Next hop IP addresses list is empty")
+	Expect(len(nextHopList)).To(BeNumerically(">=", len(podList)),
+		fmt.Sprintf("Number of speaker IP addresses[%d] is less than the number of pods[%d]",
+			len(nextHopList), len(podList)))
+
+	routesMap := make(map[string]string)
+
+	for _, frrPod := range podList {
+		if frrPod.Definition.Spec.NodeName == workerNodeList[0].Definition.Name {
+			routesMap[frrPod.Definition.Spec.NodeName] = nextHopList[1]
+		} else {
+			routesMap[frrPod.Definition.Spec.NodeName] = nextHopList[0]
+		}
+	}
+
+	return routesMap
+}
+
+func setStaticRoute(frrPod *pod.Builder, action, destIP string, nextHopMap map[string]string) (string, error) {
+	buffer, err := frrPod.ExecCommand(
+		[]string{"ip", "route", action, destIP, "via", nextHopMap[frrPod.Definition.Spec.NodeName]})
+	if err != nil {
+		if strings.Contains(buffer.String(), "File exists") {
+			glog.V(90).Infof("Warning: Route to %s already exist", destIP)
+
+			return buffer.String(), nil
+		}
+
+		if strings.Contains(buffer.String(), "No such process") {
+			glog.V(90).Infof("Warning: Route to %s already absent", destIP)
+
+			return buffer.String(), nil
+		}
+
+		return buffer.String(), err
+	}
+
+	return buffer.String(), nil
+}
+
+// DefineBGPConfigMapWithStaticRouteAndNetwork defines BGP config file with static route and network.
+func defineConfigMapWithStaticRouteAndNetwork(hubPodIPs, nodeIPAddresses []string) string {
+	bgpConfig :=
+		fmt.Sprintf("ip route %s/32 %s\n", nodeIPAddresses[1], hubPodIPs[0]) +
+			fmt.Sprintf("ip route %s/32 %s\n!\n", nodeIPAddresses[0], hubPodIPs[1])
+
+	bgpConfig += "!\nline vty\n!\nend\n"
+
+	return bgpConfig
+}
+
+func createTestPodsOnWorkers(nodeName string, portNum int) *pod.Builder {
+	testPod, err := pod.NewBuilder(
+		APIClient, "testpod-"+nodeName, tsparams.TestNamespaceName, NetConfig.CnfNetTestContainer).
+		DefineOnNode(nodeName).WithHostNetwork().WithHostPid(true).
+		RedefineDefaultCMD([]string{"/bin/bash", "-c",
+			fmt.Sprintf("testcmd -interface br-ex -protocol tcp -port %d -listen", portNum)}).
+		WithPrivilegedFlag().CreateAndWaitUntilRunning(180 * time.Second)
+	Expect(err).ToNot(HaveOccurred(), "Failed to create nginx test pod")
+
+	return testPod
+}
+
+func validateTCPTraffic(clientPod *pod.Builder, destIPAddrs []string, interfaceName,
```

Comment:
no need return an error

Reply:
please move the func to metallb.internal.cmd

Reply:
I moved it to network.internal.cmd